contrib/database/sql/internal/dsn.go

```diff
@@ -19,20 +21,36 @@ func ParseDSN(driverName, dsn string) (meta map[string]string, err error) {
 	case "mysql":
 		meta, err = parseMySQLDSN(dsn)
 		if err != nil {
+			log.Debug("Error parsing DSN for mysql: %v", err)
 			return
 		}
 	case "postgres", "pgx":
 		meta, err = parsePostgresDSN(dsn)
 		if err != nil {
+			log.Debug("Error parsing DSN for postgres: %v", err)
 			return
 		}
 	case "sqlserver":
 		meta, err = parseSQLServerDSN(dsn)
 		if err != nil {
+			log.Debug("Error parsing DSN for sqlserver: %v", err)
 			return
 		}
 	default:
-		// not supported
+		// Try to parse the DSN and see if the scheme contains a known driver name.
+		u, e := url.Parse(dsn)
+		if e != nil {
+			// dsn is not a valid URL, so just ignore
+			log.Debug("Error parsing driver name from DSN: %v", e)
+			return
```

Comment:
if you want to return here, you need to call `reduceKeys` to not change the previous behaviour (or keep the `break` statement as it was before):

```suggestion
			return reduceKeys(meta), nil
```